src/components/common/SideContent.js

```diff
@@ -59,25 +60,29 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-
-const {
-  useChildren,
-} = hooks;
+const { useItemsChildren } = hooks;
 
 const SideContent = ({ children, item }) => {
-  const isFolder = item.get('type') !== ITEM_TYPES.FOLDER;
-
-  const parentId =
-    (isFolder &&
-      getDirectParentId(item.get('path'))) ||
-    item.get('id');
   const settings = item.get('settings');
+  const isFolder = item.get('type') === ITEM_TYPES.FOLDER;
+  const { rootId } = useParams();
+
+  const parents = getParentsIdsFromPath(item.get('path') || item.get('id'));
+  const parentsIds = parents.slice(
+    parents.indexOf(rootId),
+    isFolder ? parents.length : -1,
+  );
 
-  const { data: child } = useChildren(parentId, {
+  const { data: child } = useItemsChildren([...parentsIds], {
```

Comment:
Is it to remove the immutable? You can use `.toArray()` or `.toJS()`

Reply:
It's to prevent reordering, the hashItemsIds function in query-client reorders the array by ids.
We want to keep the array in order of the split to display the pinned items